contrib/agent/src/main/java/io/opencensus/contrib/agent/AgentMain.java

```diff
@@ -57,19 +60,52 @@ public static void premain(String agentArgs, Instrumentation instrumentation) th
 
     logger.info("Initializing.");
 
-    // The classes in bootstrap.jar will be referenced from classes loaded by the bootstrap
-    // classloader. Thus, these classes have to be loaded by the bootstrap classloader, too.
+    // The classes in bootstrap.jar, such as ContextManger and ContextStrategy, will be referenced
+    // from classes loaded by the bootstrap classloader. Thus, these classes have to be loaded by
+    // the bootstrap classloader, too.
     instrumentation.appendToBootstrapClassLoaderSearch(
             new JarFile(Resources.getResourceAsTempFile("bootstrap.jar")));
 
+    checkLoadedByBootstrapClassloader(ContextManager.class);
+    checkLoadedByBootstrapClassloader(ContextStrategy.class);
+
     AgentBuilder agentBuilder = new AgentBuilder.Default()
             .disableClassFormatChanges()
             .with(AgentBuilder.RedefinitionStrategy.RETRANSFORMATION)
             .with(new AgentBuilderListener())
             .ignore(none());
-    // TODO(stschmidt): Add instrumentation for context propagation.
+    agentBuilder = LazyLoaded.addContextPropagation(agentBuilder);
+    // TODO: Add more instrumentation, potentially introducing a plugin mechanism.
     agentBuilder.installOn(instrumentation);
 
     logger.info("Initialized.");
   }
+
+  private static void checkLoadedByBootstrapClassloader(Class<?> clazz) {
+    checkState(clazz.getClassLoader() == null,
+            "%s must be loaded by the bootstrap classloader",
+            clazz);
+  }
+
+  private static class LazyLoaded {
+
+    /**
+     * Adds automatic context propagation.
```

Comment:
We try to use normal comments if needed for private methods/classes. No need to have incomplete javadoc (missing param and return).